toml/src/test/kotlin/org/rust/toml/completion/LocalCargoTomlDependencyCompletionProviderTest.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.completion
+
+import org.intellij.lang.annotations.Language
+import org.rust.ide.experiments.RsExperiments
+import org.rust.openapiext.runWithEnabledFeature
+import org.rust.toml.crates.local.CargoRegistryCrate
+import org.rust.toml.crates.local.CargoRegistryCrateVersion
+import org.rust.toml.crates.local.withMockedCrates
+
+class LocalCargoTomlDependencyCompletionProviderTest : CargoTomlCompletionTestBase() {
+    fun `test basic completion`() = doSingleCompletion("""
+        [dependencies]
+        fo<caret>
+    """, """
+        [dependencies]
+        foo = "1.0.0"<caret>
+    """, crate("foo", "1.0.0"))
+
+    fun `test hyphen-underscore normalization completion`() = doSingleCompletion("""
+        [dependencies]
+        foo-<caret>
+    """, """
+        [dependencies]
+        foo_bar = "1.0.0"<caret>
+    """, crate("foo_bar", "1.0.0"))
+
+    fun `test subwords completion`() = doSingleCompletion("""
+        [dependencies]
+        f-ba<caret>
+    """, """
+        [dependencies]
+        foo_bar = "1.0.0"<caret>
+    """, crate("foo_bar", "1.0.0"))
+
+    private fun crate(name: String, vararg versions: String): Crate =
```

Comment:
This is used also in other tests. I extracted a base test class to avoid duplication here: https://github.com/intellij-rust/intellij-rust/pull/6801/files#diff-e97b3231eba5cb633cf03dfed0b9a1ece02d7403da95fbc3390a92fe7e765615R19. However, it's inspection specific. Maybe we should introduce a kind of a "mixin" test base class which would contain the `crate` function and the `Crate` class and a function that would run the passed callback in
```kotlin
runWithEnabledFeature(RsExperiments.CRATES_LOCAL_INDEX) {
    withMockedCrates(crateMap) {
        ...
    }
}
```
?

Reply:
Introduced `LocalCargoTomlCompletionTestBase` as a completion test base for crates local index.

There still will be some code duplication between base-classes and their methods, not sure what to do with this right now

Reply:
In theory, the mixin base class could be an interface with default method implementations. Or it could be a superclass of both completion, i.e. put into some `RsTomlTestBase`. But I'm not sure if it's worth it.